### PR TITLE
fix: port to app platform [DHIS2-9388]

### PR DESCRIPTION
**Pull request by KaiVandivier**
Ports this app to the app platform, as per https://jira.dhis2.org/browse/DHIS2-9388

~~Many i18n strings were updated to the newer `i18n.t()` format from previous `d2.i18n.getTranslation()` format, except for in one place where keys received from d2-ui are being translated in `src/layout/FormFields.component.js`. Hence some legacy i18n things are existing side-by-side with the more modern version~~

~~Question: I figure I will pare down the legacy `.properties` file down to just the few strings that are translated. Do I only need to do that for the `i18n_module_en.properties` file, and the other ones will be updated by transifex?~~

~~I copied the transifex config from the Dashboards app.~~

Thanks to Médi's suggestion below, the need for legacy i18n was removed 🎉

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,65 +2,38 @@
...
-        "test-watch": "echo Everything is always ok! # mocha --reporter spec --compilers js:babel-core/register --require config/setup.js --recursive test --watch",
-        "lint": "eslint --fix src/",
-        "validate": "npm ls",
-        "prestart": "d2-manifest package.json manifest.webapp",
-        "start": "webpack-dev-server"
+        "build": "d2-app-scripts build",
+        "start": "d2-app-scripts start",
+        "test": "d2-app-scripts test",
+        "deploy": "d2-app-scripts deploy",
+        "lint": "d2-style js check && d2-style text check",
```
Comment (mediremi) [inline package.json:11]: Some extra scripts used by other apps:

```suggestion
        "lint:js": "d2-style js check",
        "lint:text": "d2-style text check",
        "lint:staged": "yarn lint:js --staged && yarn lint:text --staged",
        "lint": "yarn lint:js && yarn lint:text",
        "format:js": "d2-style js apply",
        "format:text": "d2-style text apply",
        "format:staged": "yarn format:js --staged && yarn format:text --staged",
```

**Review by mediremi — COMMENTED** [review]
`public/icon.png` should be renamed to `public/dhis2-app-icon.png` (see [docs](https://platform.dhis2.nu/#/usage/app-icon))

**Comment by mediremi** [comment]
I've tested this locally and works perfectly :tada: Good job converting all the legacy translation keys into `i18n.t` calls :+1:

File: package.json (hunk 1 above)
Comment (KaiVandivier) [reply package.json:11]: Looks good 👍

**Comment by KaiVandivier** [comment]
There are a few strings that have translations in the .po files but don't get translated in the app, for example "Property to display in analysis modules", "Male", "Female", and "Other" -- does anyone know something about that?

**Comment by KaiVandivier** [comment]
Some Transifex questions related to the port:
* Before porting, there were `<lang>.po` files and `i18n_module_<lang>.properties` files in the `i18n/module/` directory - I moved the `*.po` files to `i18n/` and the `*.properties` files to `public/i18n_legacy`, and configured the app adapter to load the legacy files from there.
* All but a few strings were converted to the `i18n.t()` format from `d2.i18n.getTranslation`, but a few of the legacy format remain because they’re coupled with the d2-ui form validators in [FormField.component.js](https://github.com/dhis2/user-profile-app/blob/7b749b1fefaeb20ca9eed0915c52d4a6b5dfe243/src/layout/FormFields.component.js#L83)
* I updated the `.tx/config` file to match the one the Dashboards app uses because it seems to use a similar setup. Does it look correct?
* Since only a few `d2.i18n.getTranslation()` strings remain, I pared down `i18n_module_en.properties` to the few remaining strings. Will the other `<lang>.properties` files get update by transifex?
* Likewise, the `<lang>.po` files are probably a little out of sync with the most recent `en.pot`. Do those also get updated by transifex?

**Comment by mediremi** [comment]
> All but a few strings were converted to the i18n.t() format from d2.i18n.getTranslation, but a few of the legacy format remain because they’re coupled with the d2-ui form validators in FormField.component.js

Through the settings app, I've been able to obtain the following translations:
```js
const translateValidatorMessage = validatorMessage => {
    switch (validatorMessage) {
        case 'value_required':
            return i18n.t('This field is required')
        case 'value_should_be_a_url':
            return i18n.t('This field should be a URL')
        case 'value_should_be_list_of_urls':
            return i18n.t('This field should contain a list of URLs')
        case 'value_should_be_a_number':
            return i18n.t('This field should be a number')
        case 'value_should_be_a_positive_number':
            return i18n.t('This field should be a positive number')
        case 'value_should_be_an_email':
            return i18n.t('This field should be an email')
    }
}
```

**Comment by KaiVandivier** [comment]
Nice, good solution 👍

**Review by mediremi — APPROVED** [review]

**Review by HendrikThePendric — APPROVED** [review]
Good work 💪 . The code seems good, the main functionality of the app is still working, and localisation is working properly too 👍 .

**Pull request merged**